tensorpack/decomposition.py

```diff
@@ -34,8 +34,27 @@ def __init__(self, data, matrix=[0], max_rr=5, method=perform_CP):
                 self.hasMatrix = True
         pass
 
-    def perform_tfac(self):
-        self.tfac = [self.method(self.data, r=rr) for rr in self.rrs]
+    def perform_tfac(self, callback=None):
+        """
+        Paramters
+        ---------
+        callback : tracker Class
+            Optional callback class to track R2X over iteration/runtime.      
+
+        Returns
+        -------
+        self.tfac : list of length max_rr
+            Each value represents the tensor factorization calculated for components 1 to max_rr.
+        self.TR2X : list of length max_rr
+            Each value represents the R2X of factorizations calculated for components 1 to max_rr.
+        self.sizeT : list of length max_rr
+            Each value represents the size of factorizations for components 1 to max_rr.
+        """
+        if callback:
+            self.tfac = [self.method(self.data, r=rr) for rr in np.delete(self.rrs, max(self.rrs)-1)]
```

Comment:
This `self` is a `Decomposition` object instead of a `tracker` one?

Reply:
This is just implementing the `tracker` object as a callback within the `Decomposition` object member function `perform_tfac()` -- it tracks the R2X of the highest component of interest.